src/std/misc/func.ss

```diff
@@ -202,3 +202,34 @@
   (if limit
     (if (> limit 0) test (lambda (_) #f))
     pred))
+
+;; pred-after-sequence returns a predicate which returns #t after a
+;; matching sequence. True is returned limit times, if limit is not #f.
+;;
+;; Example:
+;;  (filter (pred-after-sequence [1 2]) [1 2 'a 1 2 'b]) => (a b)
+(def (pred-after-sequence list (limit #f))
+  (declare (not safe) (fixnum))
+  ;; TODO when contracts merge
+  ;; (@contract (pred-sequence list limit) (or (not limit) (fixnum? limit)))
+  (unless (fixnum? limit) (set! limit -1))
+  (def i 0)
+  (def len (length list))
+  (def is-after #f)
+  (if (pair? list)
+    (lambda (v)
+      (match* (is-after limit)
+        ((#t _) (set! is-after #f) #t) ; success
+        ((_ 0)  #f)                    ; limit reached
+        (else                          ; check
+          (when (and (or (equal? (list-ref list i) v)
```

Comment:
this would be more efficient if we were `cdr`ing a partial match list in place; `list-ref` is `O(n)` while the cdring match would be `O(1)`.